### PR TITLE
Add "local linux" build script(s).

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

**Comment by dhermes** [comment]
I made this so I could play with https://github.com/google/crc32c/pull/25. When that PR is patched onto the `crc32c` submodule here, this script produces:

```
$ REPO_ROOT=$(pwd) ./scripts/local-linux/build_libcrc32c.sh
$ ls -1 usr/lib/*.so*
usr/lib/libcrc32c.so@
usr/lib/libcrc32c.so.0.1.0
$ readlink usr/lib/libcrc32c.so
libcrc32c.so.0.1.0
$ objdump -p usr/lib/libcrc32c.so | grep -n -e '\(SONAME\|NEEDED\)'
21:  NEEDED               libstdc++.so.6
22:  NEEDED               libc.so.6
23:  SONAME               libcrc32c.so.0.1.0
```

without the patch it produces:

```
$ ls -1 usr/lib/*.so*
usr/lib/libcrc32c.so
$ objdump -p usr/lib/libcrc32c.so | grep -n -e '\(SONAME\|NEEDED\)'
21:  NEEDED               libstdc++.so.6
22:  NEEDED               libc.so.6
23:  SONAME               libcrc32c.so
```

File: scripts/local-linux/build_libcrc32c.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# Check that the REPO_ROOT variable is set.
+if [[ -z "${REPO_ROOT}" ]]; then
+    echo "REPO_ROOT environment variable should be set by the caller."
+    exit 1
+fi
+
+CRC32C_INSTALL_PREFIX=${REPO_ROOT}/usr
+
+# Make sure we have an updated `pip`.
+${PY_BIN} -m pip install --upgrade pip
```
Comment (tseaver) [inline scripts/local-linux/build_libcrc32c.sh]: Shouldn't need this if we're making a virtual env (and would likely need `sudo` on most machines).

Comment (dhermes) [reply scripts/local-linux/build_libcrc32c.sh]: RE: `sudo`, very good point.

RE: "shouldn't need", I feel like it's "providing a service" to get the user the latest and greatest `pip`.

How would you feel if I let `PY_BIN` be user-configurable with a fallback to `python3.7`? This way users could do e.g.

```
PY_BIN="sudo python3.7" ./scripts/local-linux/build_libcrc32c.sh
```

Comment (tseaver) [reply scripts/local-linux/build_libcrc32c.sh]: > I feel like it's "providing a service" to get the user the latest and greatest pip.

It isn't really in scope for this script.  Can you just put in a comment that suggests upgrading pip if the script fails?

Comment (dhermes) [reply scripts/local-linux/build_libcrc32c.sh]: You mean like print something to `stderr` or just putting in a bash comment?

Comment (tseaver) [reply scripts/local-linux/build_libcrc32c.sh]: I was thinking bash comment.

Comment (dhermes) [reply scripts/local-linux/build_libcrc32c.sh]: Done.

Comment (tseaver) [reply scripts/local-linux/build_libcrc32c.sh]: I meant for the comment to replace updating the host's `pip`, which I don't believe is helpful or appropriate in a script.

Comment (dhermes) [reply scripts/local-linux/build_libcrc32c.sh]: Done

**Review by tseaver — APPROVED** [review]

**Comment by dhermes** [comment]
I'm not going to merge with my custom fork of `google/crc32c`. I can remove it and just merge the script @tseaver, WDYT?

**Comment by dhermes** [comment]
I "ported" this PR via https://github.com/googleapis/python-crc32c/commit/3aad37e1e49ca5d9672dfa1f6742d9d8d6cccae0

**Pull request closed without merging**